smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/ImportDeclarations.java

```diff
@@ -63,34 +69,41 @@ ImportDeclarations addImport(String name, String alias, String module) {
     public String toString() {
         StringBuilder result = new StringBuilder();
 
-        if (!imports.isEmpty()) {
-            for (Map.Entry<String, Map<String, String>> entry : imports.entrySet()) {
+        if (!defaultImports.isEmpty()) {
+            List<Map.Entry<String, String>> entries = new ArrayList<>(defaultImports.entrySet());
+            for (Map.Entry<String, String> importEntry : entries) {
+                result.append("import ")
+                        .append(importEntry.getValue())
+                        .append(" from \"")
+                        .append(importEntry.getKey())
+                        .append("\";\n");
+            }
+            result.append("\n");
+        }
+
+        if (!namedImports.isEmpty()) {
+            for (Map.Entry<String, Map<String, String>> entry : namedImports.entrySet()) {
                 String module = entry.getKey();
                 Map<String, String> moduleImports = entry.getValue();
-                List<Map.Entry<String, String>> nonStarEntries = new ArrayList<>(moduleImports.entrySet());
+                List<Map.Entry<String, String>> entries = new ArrayList<>(moduleImports.entrySet());
 
-                // "*" imports must be special-cased and can't be joined with named imports, and
-                // remove "*" imports from nonStarEntries which is iterated afterwards.
+                // "*" imports are not supported https://github.com/awslabs/smithy-typescript/issues/211
                 for (Map.Entry<String, String> importEntry : moduleImports.entrySet()) {
                     if (importEntry.getValue().equals("*")) {
-                        nonStarEntries.remove(importEntry);
-                        result.append("import ")
-                                .append(createImportStatement(importEntry))
-                                .append(" from \"")
-                                .append(module)
-                                .append("\";\n");
+                        throw new CodegenException("Star imports are not supported, attempted for " + module
+                                + ". Use default import instead.");
                     }
                 }
 
-                if (nonStarEntries.size() == 1) {
+                if (entries.size() == 1) {
                     result.append("import { ")
-                            .append(createImportStatement(nonStarEntries.get(0)))
+                            .append(createImportStatement(entries.get(0)))
```

Comment:
If you convert this to a `Set`, then this becomes, `entries.iterator().next()`.

Reply:
Updated in [`3f34b04` (#212)](https://github.com/awslabs/smithy-typescript/pull/212/commits/3f34b04375fc742c1e22e44ab953f43f81bb559a)